### ISSUE TITLE
Simplify organization of files in the frontend

### ISSUE THREAD

**Issue by abidlabs** [gradio-app/gradio#5763]
Right now, there's a lot of code repetition, which makes it cumbersome to e.g. modify props when building custom components

The current file structure includes separate directories for InteractiveComponent, StaticComponent, etc. It would be conceptually simpler to have a single file per component that includes both the interactive and static versions. 

cc @pngwn

**Comment by henryruhs** [comment]
In FaceFusion we have [components](https://github.com/facefusion/facefusion/tree/master/facefusion/uis/components) per file that can be reused in different [layouts](https://github.com/facefusion/facefusion/tree/master/facefusion/uis/layouts).

## default

![01](https://github.com/gradio-app/gradio/assets/1835397/7404baaf-13b1-464b-948c-9ba76137d5cd)

Source: https://github.com/facefusion/facefusion/blob/master/facefusion/uis/layouts/benchmark.py

## benchmark

![02](https://github.com/gradio-app/gradio/assets/1835397/b0fc3096-476e-4b69-aa5f-b0bf83be6fe4)

Source: https://github.com/facefusion/facefusion/blob/master/facefusion/uis/layouts/default.py

## events

In terms of event handling there is a simple `register_component` and `get_component` using a shared dict.

### register component

```
def render() -> None:
	global BENCHMARK_RUNS_CHECKBOX_GROUP
	global BENCHMARK_CYCLES_SLIDER

	BENCHMARK_RUNS_CHECKBOX_GROUP = gradio.CheckboxGroup(
		label = wording.get('benchmark_runs_checkbox_group_label'),
		value = list(BENCHMARKS.keys()),
		choices = list(BENCHMARKS.keys())
	)
	ui.register_component('benchmark_runs_checkbox_group', BENCHMARK_RUNS_CHECKBOX_GROUP)
```

### get component to set handler

```
def listen() -> None:
	benchmark_runs_checkbox_group = ui.get_component('benchmark_runs_checkbox_group')
	if benchmark_runs_checkbox_group:
		BENCHMARK_START_BUTTON.click(start, inputs = benchmark_runs_checkbox_group, outputs = BENCHMARK_RESULTS_DATAFRAME)
```

## concept

A strict separation of concerns using `render` and `listen` make this possible.

**Comment by abidlabs** [comment]
This is closed now as part of 4.0 work, thanks @pngwn1
